spotbugs/src/main/java/edu/umd/cs/findbugs/detect/DontUseFloatsAsLoopCounters.java

```diff
@@ -6,23 +6,104 @@
 import edu.umd.cs.findbugs.BugReporter;
 import edu.umd.cs.findbugs.StatelessDetector;
 import edu.umd.cs.findbugs.bcel.OpcodeStackDetector;
-import java.util.stream.Stream;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
 
 public class DontUseFloatsAsLoopCounters extends OpcodeStackDetector implements StatelessDetector {
 
     private final BugReporter bugReporter;
 
+    private static final Map<Integer, Integer> FLOAT_LOADERS = new HashMap<>();
```

Comment:
Perhaps initialise this inside the `static` block, with a `Collections.unmodifiableMap`?